Library/Homebrew/test/livecheck/livecheck_spec.rb

```diff
@@ -6,15 +6,21 @@
 describe Homebrew::Livecheck do
   subject(:livecheck) { described_class }
 
+  CASK_URL = "https://brew.sh/test-0.0.1.dmg"
+  HEAD_URL = "https://github.com/Homebrew/brew.git"
+  HOMEPAGE_URL = "https://brew.sh"
+  LIVECHECK_URL = "https://formulae.brew.sh/api/formula/ruby.json"
+  STABLE_URL = "https://brew.sh/test-0.0.1.tgz"
```

Comment:
Use `let` instead of constants.

Reply:
I agree that `let` is more appropriate. I was using constants because a string that's defined using `let` isn't available in the `formula` and `cask` blocks. When using a constant, `url STABLE_URL` works in the `f` formula. When using `let(:stable_url) { ... }`, `url stable_url` gives an ``` undefined local variable or method `stable_url'``` error.

My goals were to reduce repetition of string literals in multiple places and to make the tests easier to read (i.e., it's somewhat ambiguous as to what "https://brew.sh/test-0.0.1.tgz" references but `STABLE_URL` is clear).

---

I've updated this to use string literals in the formula/cask definitions and `let` to define strings that can be referenced in tests (so we can reference something that's readily understandable like `stable_url` instead of an ambiguous string literal).

I'm considering trying to extract the various formulae/casks that we define in livecheck's tests into separate files (like test/support/fixtures/cask/Casks) sometime in the future. If we go down that path, we would have to lose the constants anyway, so I'm good with the `let` approach here.